master/internal/api_project.go

```diff
@@ -53,45 +53,148 @@ func (a *apiServer) getProjectColumnsByID(
+			Type:        projectv1.ColumnType_COLUMN_TYPE_TEXT, // is this right?
+		},
+		{
+			Column:      "forked",
+			DisplayName: "Forked",
+			Location:    projectv1.LocationType_LOCATION_TYPE_EXPERIMENT,
+			Type:        projectv1.ColumnType_COLUMN_TYPE_NUMBER,
+		},
+		{
+			Column:      "start_time",
+			DisplayName: "Start time",
+			Location:    projectv1.LocationType_LOCATION_TYPE_EXPERIMENT,
+			Type:        projectv1.ColumnType_COLUMN_TYPE_DATE,
+		},
+		{
+			Column:      "duration",
+			DisplayName: "Duration",
+			Location:    projectv1.LocationType_LOCATION_TYPE_EXPERIMENT,
+			Type:        projectv1.ColumnType_COLUMN_TYPE_NUMBER, // ?
+		},
+		{
+			Column:      "num_trials",
+			DisplayName: "Trial count",
+			Location:    projectv1.LocationType_LOCATION_TYPE_EXPERIMENT,
+			Type:        projectv1.ColumnType_COLUMN_TYPE_NUMBER,
+		},
+		{
+			Column:      "state",
+			DisplayName: "State",
+			Location:    projectv1.LocationType_LOCATION_TYPE_EXPERIMENT,
+			Type:        projectv1.ColumnType_COLUMN_TYPE_TEXT,
+		},
+		{
+			Column:      "searcher_type",
+			DisplayName: "Searcher type",
+			Location:    projectv1.LocationType_LOCATION_TYPE_EXPERIMENT,
+			Type:        projectv1.ColumnType_COLUMN_TYPE_TEXT,
+		},
+		{
+			Column:      "resource_pool",
+			DisplayName: "Resource pool",
+			Location:    projectv1.LocationType_LOCATION_TYPE_EXPERIMENT,
+			Type:        projectv1.ColumnType_COLUMN_TYPE_TEXT,
+		},
+		{
+			Column:      "progress",
+			DisplayName: "Progress",
+			Location:    projectv1.LocationType_LOCATION_TYPE_EXPERIMENT,
+			Type:        projectv1.ColumnType_COLUMN_TYPE_NUMBER,
+		},
+		{
+			Column:      "checkpoint_size",
+			DisplayName: "Checkpoint size",
+			Location:    projectv1.LocationType_LOCATION_TYPE_EXPERIMENT,
+			Type:        projectv1.ColumnType_COLUMN_TYPE_NUMBER,
+		},
+		{
+			Column:      "checkpoint_count",
+			DisplayName: "Checkpoint count",
+			Location:    projectv1.LocationType_LOCATION_TYPE_EXPERIMENT,
+			Type:        projectv1.ColumnType_COLUMN_TYPE_NUMBER,
+		},
+		{
+			Column:      "username",
+			DisplayName: "User",
+			Location:    projectv1.LocationType_LOCATION_TYPE_EXPERIMENT,
+			Type:        projectv1.ColumnType_COLUMN_TYPE_TEXT,
+		},
+	}
+
+	hyperparameters := []struct {
+		Hyperparameter string
 	}{}
-	err := db.Bun().
-		NewSelect().Table("projects").Column("hyperparameters").Where(
-		"id = ?", id).Scan(ctx, &hyperparameters)
+
+	// Get hyperpatameters columns
+	err := db.Bun().NewSelect().
+		ColumnExpr("jsonb_array_elements_text(hyperparameters) as hyperparameter").
+		Table("projects").
+		Where("id = ?", id).
+		Scan(ctx, &hyperparameters)
+
 	if err != nil {
 		return nil, err
 	}
+	for _, h := range hyperparameters {
+		columns = append(columns, &projectv1.ProjectColumn{
+			Column:   h.Hyperparameter,
+			Location: projectv1.LocationType_LOCATION_TYPE_HYPERPARAMETERS,
+			Type:     projectv1.ColumnType_COLUMN_TYPE_UNSPECIFIED,
```

Comment:
Is it possible to add specific type for each hyperparameter value? Without types, its hard to handle values for table filter

Reply:
see previous discussion [here](https://github.com/determined-ai/determined/pull/6563#discussion_r1167302019) -- in short, yes, but because the erd points to wanting to sort/filter on the experiment config and not the resolved hyperparameter on the trial, it might not be what the endpoint is actually operating on?